### ISSUE TITLE
[Bug]: Transpilation Bug with Decorators - Incorrectly Passing 'type: parameter' Instead of 'type: class'

### ISSUE THREAD

**Issue by Technologeek** [babel/babel#16197]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

babel-loader (webpack)

### Input code

[Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=AIMw9mBQDGA2CGBnRACAKgU0QFxQbwF9JIQBXAO2mwEsxyVwwAKbAc2wBoUqAPASnyQU3OojCwMAOlhhWTAETYAngAcMKaonldekgNbVyAEz4BuSEUhA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.23.6&externalPlugins=&assumptions=%7B%7D)

### Configuration

````js
 {
					loader: "babel-loader",
					options: {
						presets: [
							[
								"@babel/preset-env",
								{
									targets: "> 1%, not dead",
									loose: true
								}
							],
							"@babel/preset-react",
							"@babel/preset-typescript"
						],
						plugins: [
							["@babel/plugin-proposal-decorators", { version: "2023-05" }],
							["babel-plugin-styled-components", { ssr: false }],
							["@babel/plugin-transform-class-static-block",{ loose: false }],
						]
					}
				}
````
### Current and expected behavior

We've encountered a transpilation issue when using Babel with MobX decorators. After downgrading due to a separate issue, we observed a new problem related to the Babel configuration and MobX decorators. The transpiled code incorrectly passes type: parameter instead of type: class into the decorator, affecting not just MobX but other decorators as well.

**Additional Information**
We have consulted the [MobX migration guide](https://mobx.js.org/migrating-from-4-or-5.html#getting-started), but found that the advice is outdated and not applicable to our scenario.

The issue seems to be with Babel's handling of decorators rather than MobX itself, as it affects other decorators too.

Expected Behavior
The transpiled code should correctly pass type: class into the decorator, as per the standard decorator behavior.

### Environment

- System:
    OS: Linux 5.15 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
 -  Binaries:
    Node: 20.10.0 
    Yarn: 4.0.2 
    npm: 9.8.1 
  - Monorepos:
    Yarn Workspaces: 4.0.2
  - npm Packages:
    eslint: ^8.47.0 => 8.56.0 
    jest: ^27.0.6 => 27.5.1

**Comment by JLHwung** [comment]
Duplicate of #16179 

We will do a patch release soon.

**Comment by JLHwung** [comment]
Fixed in v7.23.7.
